lib/elixir/src/elixir_erl_compiler.erl

```diff
@@ -135,6 +135,12 @@ custom_format(sys_core_fold, nomatch_guard) ->
   "this check/guard will always yield the same result";
 
 %% Handle literal eval failures
+custom_format(sys_core_fold, {eval_failure, {_, {Name, Arity}}, Error}) ->
```

Comment:
This will unfortunately show functions according to Erlang operators, import rules, and function names.

You need to match on:

* `{operator, {Name, Arity}}`
* `{function, {Name, Arity}}`
* `{function, {Module, Name, Arity}}`

And then translate it back to Elixir operators and function names. You can use `elixir_rewrite:erl_to_ex/3` for this but I have also asked for this to be simplified so we don't have to work around Erlang's rules: https://github.com/erlang/otp/commit/8ecc648df22dbfe9dd68a3e2b968ba2b6a2ca7fe#r46899624.

Reply:
make sense, instead of `{operator, {Name, Arity}}`, erl compiler would return `{Module, Name, Arity}`, right?

In fact, I think the compiler should return `{Module, Name, Args}` (not `Arity`) so that we can pass it to `elixir_rewrite:erl_to_ex/3`.

Reply:
I've created a PR https://github.com/erlang/otp/pull/3051 so that the compiler emits `{eval_failure, {Mod, Name, Arity}}`. It emits Arity instead of Args because of possible performance issues.